server/medical-contact/collection.ts

```diff
@@ -0,0 +1,124 @@
+import type {HydratedDocument, Types} from 'mongoose';
+import type {MedicalContact} from './model';
+import MedicalContactModel from './model';
+import AccountCollection from '../account/collection';
+
+class MedicalContactCollection {
+  /**
+   * Add a new medical contact
+   *
+   * @param {string} ownerId - The id of the owner of the medical contact
+   * @param {string} title - The username of the user
+   * @param {string} first_name - The username of the user
+   * @param {string} last_name - The username of the user
+   * @param {string} hospital - The username of the user
+   * @param {string} specialty - The username of the user
+   * @param {string} phone_number - The username of the user
+   * @param {string} notes - The username of the user
+   * @return {Promise<HydratedDocument<MedicalContact>>} - The newly created medical contact
+   */
+  static async addOne(ownerId: Types.ObjectId | string, title: string, first_name: string, last_name: string, hospital: string, specialty: string, phone_number: string, notes: string): Promise<HydratedDocument<MedicalContact>> {
+    
+    const active = true; // Since medical contact was just created, it is active
+    const medicalContact = new MedicalContactModel({
+        ownerId,
+        active, 
+        title, 
+        first_name, 
+        last_name, 
+        hospital, 
+        specialty, 
+        phone_number, 
+        notes
+    });
+    await medicalContact.save(); // Saves user to MongoDB
+    return medicalContact.populate('ownerId');
+  }
+
+  /**
+   * Find a medical contact by medicalContactId.
+   *
+   * @param {string} medicalContactId - The id of the medical contact to find
+   * @return {Promise<HydratedDocument<Account>> | Promise<null>} - The medical contact with the given id, if any
+   */
+   static async findOne(medicalContactId: Types.ObjectId | string): Promise<HydratedDocument<MedicalContact>> {
+    return MedicalContactModel.findOne({_id: medicalContactId}).populate('ownerId');
+  }
+
+  /**
+   * Get all the medical contacts in by given owner
+   *
+   * @param {string} ownerId - The id of the owner of the medical contacts
+   * @return {Promise<HydratedDocument<MedicalContact>[]>} - An array of all of the medical contacts sorted in alphabetical order by last name
+   */
+   static async findAllByOwnerId(ownerId: string): Promise<Array<HydratedDocument<MedicalContact>>> {
+    const owner = await AccountCollection.findOneByAccountId(ownerId);
```

Comment:
I think this call is superfluous. Could just use the passed in owner id in the next line.

Reply:
In my model I keep ownerId as a Schema.Types.ObjectId, so I think I need to keep this call

Reply:
have you tried it without? For example in your `findOne` above you take in a string id and search for an `ObjectId` in the database. I'm pretty sure `ObjectId` is just a constrained string with some extra semantics.

Reply:
I feel like I ran into a problem during my Fritter A5 – for now, I'll just comment out the line and fix the line below to just use the ownerId param (and worst case, I can go back in and fix it)